0022-circuit-block.md

```diff
@@ -0,0 +1,355 @@
+To make the user intent clearer here, we propose adding a `QuantumCircuit.noop` to make the intent explicit:
+
+```python
+from qiskit.circuit import QuantumCircuit
+qc = QuantumCircuit(3, 3)
+with qc.box():
+    # This `box` will be full-width on the circuit, despite every qubit and clbit undergoing the identity.
+    qc.noop(*qc.qubits, *qc.clbits)
+```
+
+The `noop` instruction will do the standard Qiskit circuit error checking to validate its inputs are valid resources for the circuit, but will not cause a new entry in the instruction list.
+The control-flow builders will still pick this up as a "used" resource.
+
+
+### Transpiler semantics of `box`
+
+The transpiler must fail if the backend reports that it cannot handle the `box` instruction, as for other control-flow operations.
+In the absence of any annotations, the transpiler must retain any boxes in the input circuit, and the content of the box must be transpiled up to the same equivalent and hardware-compatibility guarantees that the rest of the tranpsiler guarantees, as if the boxed content was a stand-alone circuit (e.g. a unitary operation must have the same action before and after compilation, up to layout and routing effects, and all gates must be ISA supported).
+
+Transpiler passes are permitted to insert new boxes around instruction groups.
+A `box` may only be removed (and replaced by some representation of its body) by a transpiler pass if the annotations on the `box` permit it (see below).
+The default should be to assume that the `box` cannot be removed.
+
+
+## Annotations on `box`
+
+In the above, we have talked about annotations on the `box`.
+This is by analogy to the OpenQASM 3 concept.
+In more generic terms, an annotation will be
+
+> a compiler directive applied in a limited scope
+
+For this RFC, we only consider annotations on `box`es, but in the future, this may be expanded to other general instructions.
+
+Annotations may affect how the scope of the body is compiled.
+For example, an annotation may mark that one `box` should be scheduled using an ALAP schedule internally, regardless of how the outer scope of the circuit is being scheduled.
+Such an annotation may live inside Qiskit SDK.
+
+Other use cases for `box` may have downstream transpiler passes introduce new `box`es with annotations that may be consumed by some later point in their compilation stack.
+For example, IBM's primitives implementations may want to create blocks around blocks of instructions that will undergo noise learning as a single entity, and annotated these boxes with identifiers and strategies around the noise learning, for consumption by a lower part of its own stack.
+Qiskit SDK must provide a framework such that transpiler passes can function in the presence of annotations that they don't necessarily understand entirely.
+
+> [!NOTE]
+> This section is definitely not complete yet; the text of it just represents a starting point for the thinking.
+
+### Semantics of custom
```

Comment:
This is meant to say "custom annotations".  The section no being fully formalised doesn't need to hold up the rest of the RFC - a fair amount of this will be Qiskit-specific design.  It will matter to pass authors, but large amounts of it are going to be about what's technically feasible to Qiskit, especially for a first MVP, and we might need to evolve a bit from there.